Add circuit sapling react app

eloaverona opened the pull request:
This PR:
- Sets ups a docker network for demos and test.
- Adds a basic react app for the circuit sapling

#### To test:
1.  Run:
```
docker-compose -f docker/docker-compose.yaml up --build
```
2. Go to `http://localhost:3030/circuits`:
<img width="1256" alt="Screen Shot 2020-04-08 at 9 06 37 AM" src="https://user-images.githubusercontent.com/14094978/78793633-49d5b480-7978-11ea-839f-d6f12f554b28.png">

Commit:
Fix app dockerfile

The dockerfile was referencing an outdated directory structure. 

Signed-off-by: Eloá Franca Verona <eloafran@bitwise.io>

Commit:
Add docker-compose file defining a demo network

Signed-off-by: Eloá Franca Verona <eloafran@bitwise.io>

Commit:
Add circuit sapling react app

Adds a basic react app for circuit sapling

Signed-off-by: Eloá Franca Verona <eloafran@bitwise.io>

Commit:
Add circuit sapling to sapling-dev-server

Signed-off-by: Eloá Franca Verona <eloafran@bitwise.io>

dplumb94 commented on docker/docker-compose.yaml:
```diff
@@ -0,0 +1,123 @@
+  contracts:
+  key-registry:
+
+services:
```
This looks good but we might want to add another node so we can test both parties proposing/voting on a circuit.

eloaverona replied:
I added a second node, ui and db.

dplumb94 commented on sapling-dev-server/Dockerfile:
```diff
@@ -21,6 +21,10 @@ RUN npm config set unsafe-perm true
 
+RUN cd /saplings/circuits && \
+  npm install && \
+  yarn deploy
```
I believe it is this way in the Grid saplings, but to be consistent you probably want to do `npm run deploy`. We had to switch to npm due to git dependencies not working properly with yarn.

eloaverona replied:
Updated

dplumb94 commented on sapling-dev-server/userSaplings:
```diff
@@ -1 +1,14 @@
+      "localhost:3030/sapling-dev-server/circuits/css/circuits.css"
+    ],
+    "workerFiles": [],
+    "icon": "https://via.placeholder.com/48/333333/FFFFFF?text=C"
```
I think we should skip the placeholder. My browser is blocking it now too. Davey might have an icon we can use instead. I have a PR up in grid that shows how to include an icon in the sapling dev server.

eloaverona replied:
Addressed

Commit:
Fix build for linting and test images

The image defined in app/Dockerfile was updated
and cannot be used for linting or run tests. This adds a new image
that can be used for both.

Signed-off-by: Eloá Franca Verona <eloafran@bitwise.io>

agunde406 reviewed (approved):
Tested locally
